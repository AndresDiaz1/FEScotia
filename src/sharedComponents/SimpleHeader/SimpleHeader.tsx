import React from 'react';
import { IonButton, IonGrid, IonHeader, IonLabel, IonRow, IonToolbar } from '@ionic/react';
import { clearDB } from '../../data/mainDB';
import { useDispatch } from 'react-redux';
import { logout } from '../../login/LoginStore';
import { PATHS } from '../../AppConstants';
import { useHistory } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';

interface HeaderProps {
  title: string;
  goBack: boolean;
}

const SimpleHeader: React.FC<HeaderProps> = ({ title, goBack }) => {
  const dispatch = useDispatch();
  const history = useHistory();

  const doLogout = async()=> {
    await clearDB()
    dispatch(logout());
    history.push(PATHS.LOGIN);
  }

  return (
    <IonHeader>
      <IonToolbar color="scotia-red">
        <IonGrid>
          <IonRow className="ion-justify-content-between">
            {
              goBack ? 
              <FontAwesomeIcon className="go-back-icon" icon={faAngleLeft} onClick={()=>history.goBack()} size="lg"/>
              :
              <></>
            }
            <IonLabel>{title}</IonLabel>
            <IonButton className="logout-button" size="small" color="light" shape="round" fill="outline" onClick={()=>doLogout()}>Logout</IonButton>
          </IonRow>
        </IonGrid>
      </IonToolbar>
    </IonHeader>
  );
};

export default SimpleHeader;
