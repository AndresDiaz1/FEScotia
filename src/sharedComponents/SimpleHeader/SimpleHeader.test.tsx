import React from 'react';
import { mount } from 'enzyme';
import SimpleHeader from './SimpleHeader';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

const mockHistoryGoBack = jest.fn();
const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    goBack: mockHistoryGoBack,
    push: mockHistoryPush,
  }),
}));

describe('SimpleHeader Component', () => {
  describe('Initial state', () => {
    it('Should render', async() => {
      const account = {
        id: '1',
        clientId: 1,
        accountType: 'savings',
        status: 60,
        availableValue: 33,
      };
      const state = {
        accounts: {
          isLoading: false,
          accounts: [account],
          errorMessage: '',
        },
      };

      const store = configureStore([thunk])(state);

      const component = mount(
        <Provider store={store}>
         <SimpleHeader title="Simple Header Title" goBack={false} />
         </Provider>
      );

      expect(component).toMatchSnapshot();
    });
    it('Should render with back button', () => {
      const account = {
        id: '1',
        clientId: 1,
        accountType: 'savings',
        status: 60,
        availableValue: 33,
      };
      const state = {
        accounts: {
          isLoading: false,
          accounts: [account],
          errorMessage: '',
        },
      };

      const store = configureStore([thunk])(state);

      const component = mount(
        <Provider store={store}>
         <SimpleHeader title="Simple Header Title" goBack={true} />
         </Provider>
      );
      const goBackArrow = component.find('FontAwesomeIcon.go-back-icon');
      goBackArrow.simulate('click');
      expect(mockHistoryGoBack).toHaveBeenCalled();
      expect(component).toMatchSnapshot();
    });
  });
});
