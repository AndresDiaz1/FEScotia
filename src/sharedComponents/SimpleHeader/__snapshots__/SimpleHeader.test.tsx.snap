// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleHeader Component Initial state Should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SimpleHeader
    goBack={false}
    title="Simple Header Title"
  >
    <ForwardRef(IonHeader)>
      <IonHeader
        forwardedRef={null}
      >
        <ion-header>
          <ForwardRef(IonToolbar)
            color="scotia-red"
          >
            <IonToolbar
              color="scotia-red"
              forwardedRef={null}
            >
              <ion-toolbar
                color="scotia-red"
              >
                <ForwardRef(IonGrid)>
                  <IonGrid
                    forwardedRef={null}
                  >
                    <ion-grid>
                      <ForwardRef(IonRow)
                        className="ion-justify-content-between"
                      >
                        <IonRow
                          className="ion-justify-content-between"
                          forwardedRef={null}
                        >
                          <ion-row>
                            <ForwardRef(IonLabel)>
                              <IonLabel
                                forwardedRef={null}
                              >
                                <ion-label>
                                  Simple Header Title
                                </ion-label>
                              </IonLabel>
                            </ForwardRef(IonLabel)>
                            <ForwardRef(IonButton)
                              className="logout-button"
                              color="light"
                              fill="outline"
                              onClick={[Function]}
                              shape="round"
                              size="small"
                            >
                              <IonButton
                                className="logout-button"
                                color="light"
                                fill="outline"
                                forwardedRef={null}
                                onClick={[Function]}
                                shape="round"
                                size="small"
                              >
                                <ion-button
                                  color="light"
                                  fill="outline"
                                  onClick={[Function]}
                                  shape="round"
                                  size="small"
                                >
                                  Logout
                                </ion-button>
                              </IonButton>
                            </ForwardRef(IonButton)>
                          </ion-row>
                        </IonRow>
                      </ForwardRef(IonRow)>
                    </ion-grid>
                  </IonGrid>
                </ForwardRef(IonGrid)>
              </ion-toolbar>
            </IonToolbar>
          </ForwardRef(IonToolbar)>
        </ion-header>
      </IonHeader>
    </ForwardRef(IonHeader)>
  </SimpleHeader>
</Provider>
`;

exports[`SimpleHeader Component Initial state Should render with back button 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SimpleHeader
    goBack={true}
    title="Simple Header Title"
  >
    <ForwardRef(IonHeader)>
      <IonHeader
        forwardedRef={null}
      >
        <ion-header>
          <ForwardRef(IonToolbar)
            color="scotia-red"
          >
            <IonToolbar
              color="scotia-red"
              forwardedRef={null}
            >
              <ion-toolbar
                color="scotia-red"
              >
                <ForwardRef(IonGrid)>
                  <IonGrid
                    forwardedRef={null}
                  >
                    <ion-grid>
                      <ForwardRef(IonRow)
                        className="ion-justify-content-between"
                      >
                        <IonRow
                          className="ion-justify-content-between"
                          forwardedRef={null}
                        >
                          <ion-row>
                            <FontAwesomeIcon
                              border={false}
                              className="go-back-icon"
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    256,
                                    512,
                                    Array [],
                                    "f104",
                                    "M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z",
                                  ],
                                  "iconName": "angle-left",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              onClick={[Function]}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size="lg"
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-angle-left fa-w-8 fa-lg go-back-icon"
                                data-icon="angle-left"
                                data-prefix="fas"
                                focusable="false"
                                onClick={[Function]}
                                role="img"
                                style={Object {}}
                                viewBox="0 0 256 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                            <ForwardRef(IonLabel)>
                              <IonLabel
                                forwardedRef={null}
                              >
                                <ion-label>
                                  Simple Header Title
                                </ion-label>
                              </IonLabel>
                            </ForwardRef(IonLabel)>
                            <ForwardRef(IonButton)
                              className="logout-button"
                              color="light"
                              fill="outline"
                              onClick={[Function]}
                              shape="round"
                              size="small"
                            >
                              <IonButton
                                className="logout-button"
                                color="light"
                                fill="outline"
                                forwardedRef={null}
                                onClick={[Function]}
                                shape="round"
                                size="small"
                              >
                                <ion-button
                                  color="light"
                                  fill="outline"
                                  onClick={[Function]}
                                  shape="round"
                                  size="small"
                                >
                                  Logout
                                </ion-button>
                              </IonButton>
                            </ForwardRef(IonButton)>
                          </ion-row>
                        </IonRow>
                      </ForwardRef(IonRow)>
                    </ion-grid>
                  </IonGrid>
                </ForwardRef(IonGrid)>
              </ion-toolbar>
            </IonToolbar>
          </ForwardRef(IonToolbar)>
        </ion-header>
      </IonHeader>
    </ForwardRef(IonHeader)>
  </SimpleHeader>
</Provider>
`;
